test/test_buffer.cpp

```diff
@@ -350,6 +352,156 @@ namespace sparrow
             buffer_test_type b3(make_test_buffer(size2), size2);
             CHECK(b1 != b3);
         }
+
+        TEST_CASE("emplace")
```

Comment:
All the insertion operations needs to be tested with these conditions:
- when the container is empty (just constructed)
- when it has values
  - insert at the beginning (the only one tested here for example)
  - insert in the middle
  - insert at the end